### PR TITLE
tell IDEA to treat generated LogstashProtos.java as a source

### PR DESCRIPTION
**Pull request by jameshfisher**

### CODE REVIEW

File: logstash-commons/build.gradle [hunk 1]
```diff
@@ -13,23 +13,21 @@ dependencies {
...
-        sourceDirs  += file('build/generated/source/proto')
+        excludeDirs -= file("$buildDir")
+        excludeDirs += file("$buildDir/classes")
+        excludeDirs += file("$buildDir/dependency-cache")
+        excludeDirs += file("$buildDir/libs")
+        excludeDirs += file("$buildDir/reports")
+        excludeDirs += file("$buildDir/resources")
+        excludeDirs += file("$buildDir/tmp")
+
+        sourceDirs  += file("$buildDir/generated/source/proto")
```
Comment (mwl) [inline logstash-commons/build.gradle]: I think it should be `$buildDir/generated/source/proto/main/java`?

Comment (jhftrifork) [inline logstash-commons/build.gradle]: It does seem to work as it is, but by comparison with the standard `src/main/java`, I think you're right

Comment (jhftrifork) [inline logstash-commons/build.gradle]: changed

**Comment by mwl** [comment]
Doesn't build much
```
gw clean compile
Using gradle at '/Users/mwl/IdeaProjects/es-mesos/logstash/gradlew' to run buildfile '/Users/mwl/IdeaProjects/es-mesos/logstash/build.gradle':

Current version: 0.0.6

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/mwl/IdeaProjects/es-mesos/logstash/logstash-commons/build.gradle' line: 16

* What went wrong:
A problem occurred evaluating project ':logstash-commons'.
> Could not find method idea() for arguments [build_bf9dm5h8za2ii9pix8y2qd7lf$_run_closure4@52621501] on project ':logstash-commons'.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED
```

**Comment by mwl** [comment]
Looks like it's a `apply plugin: 'idea'` that's needed in the beginning of the file

**Comment by jhftrifork** [comment]
@mwl done

**Pull request merged**